src/stdlib_linalg.fypp

```diff
@@ -214,6 +221,94 @@ module stdlib_linalg
     #:endfor
   end interface is_hessenberg
 
+
+  interface det
+    !!### Summary 
+    !! Interface for computing matrix determinant.
+    !!
+    !!### Description
+    !! 
+    !! This interface provides methods for computing the determinant of a matrix.
+    !! Supported data types include real and complex.
+    !! 
+    !!@note The provided functions are intended for square matrices.          
+    !! 
+    !!### Example
+    !!
+    !!```fortran
+    !!
+    !!    real(sp) :: a(3,3), d
+    !!    type(linalg_state_type) :: state  
+    !!    a = reshape([1, 2, 3, 4, 5, 6, 7, 8, 9], [3, 3])
+    !!    
+    !!    d = det(a,err=state)
+    !!    if (state%ok()) then 
+    !!       print *, 'Success! det=',d
+    !!    else
+    !!       print *, state%print()
+    !!    endif
+    !!
+    !!```
+    !!     
+    #:for rk,rt in RC_KINDS_TYPES
+    #:if rk!="xdp"
```

Comment:
[504d90d](https://github.com/fortran-lang/stdlib/pull/798/commits/504d90d8132136246e88bef23240ad6f8b2b92a0)

Reply:
Even though it is not automated generation, should we manually implement blas and lapack backends for `xdp`? They would be the same as for quad-precision, because all constants are generalized. We should just decide about real/complex prefixes, like `s/c` vs. `d/z` vs. `q/w`, for xdp they may be `x/y`?

Reply:
Supporting `xdp` would be a nice feature IMO (even if I never use `xdp`). `x/y` prefixes make sense to me. If you think it is easy to implement it, I would  support it. However, I don't think it is a priority right now.

Reply:
If it serves for the discussion, I'm using the following:
in the `common.fypp` I added the notion of SUFFIX like
```
...
#! Collected (kind, type, suffix) tuples for real types
#:set REAL_KINDS_TYPES = list(zip(REAL_KINDS, REAL_TYPES, REAL_KINDS))

#! Complex types to be considered during templating
#:set CMPLX_SUFFIX = ["c{}".format(k) for k in CMPLX_KINDS]

#! Collected (kind, type, suffix) tuples for complex types
#:set CMPLX_KINDS_TYPES = list(zip(CMPLX_KINDS, CMPLX_TYPES, CMPLX_SUFFIX))
```
Which then allows my to do this in the `.fypp`
```Fortran
#:for k1, t1, s1 in (KINDS_TYPES)
    type, public, extends(COO_t) :: COO_${s1}$
    ${t1}$, allocatable :: data(:) 
contains
     procedure :: get => get_value_coo_${s1}$
     procedure :: set => set_value_coo_${s1}$
end type
#:endfor
```
and obtaining this in the fortran file:
```Fortran
type, public, extends(COO_t) :: COO_sp
        real(sp), allocatable :: data(:) 
    contains
        procedure :: get => get_value_coo_sp
        procedure :: set => set_value_coo_sp
    end type
    type, public, extends(COO_t) :: COO_dp
        real(dp), allocatable :: data(:) 
    contains
        procedure :: get => get_value_coo_dp
        procedure :: set => set_value_coo_dp
    end type
    type, public, extends(COO_t) :: COO_csp
        complex(sp), allocatable :: data(:) 
    contains
        procedure :: get => get_value_coo_csp
        procedure :: set => set_value_coo_csp
    end type
    type, public, extends(COO_t) :: COO_cdp
        complex(dp), allocatable :: data(:) 
    contains
        procedure :: get => get_value_coo_cdp
        procedure :: set => set_value_coo_cdp
    end type
```
So basically considering the real types as reference, and the complex types as extensions adding the `c` in front of the suffix.